service/api_task.go

```diff
@@ -182,14 +182,19 @@ func (as *APIServer) EndTask(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// mark task as finished
-	updates := model.StatusChanges{}
-	err = model.MarkEnd(t, APIServerLockTitle, finishTime, details, projectRef.DeactivatePrevious, &updates)
+	err = model.MarkEnd(t, APIServerLockTitle, finishTime, details, projectRef.DeactivatePrevious)
 	if err != nil {
 		err = errors.Wrapf(err, "Error calling mark finish on task %v", t.Id)
 		as.LoggedError(w, r, http.StatusInternalServerError, err)
 		return
 	}
 
+	job := units.NewUpdateBuildAndVersionJob(t.Id)
+	if err := evergreen.GetEnvironment().RemoteQueue().Put(job); err != nil {
```

Comment:
I think there's a queue in the api server struct that you should use

Reply:
done